src/lib.rs

```diff
@@ -307,23 +307,43 @@ pub trait Resampler<T>: Send {
     /// the buffer size is only guaranteed to be sufficient to prevent allocation
     /// within [Resampler::process_into_buffer] until the resampling ratio is changed.
     fn allocate_output_buffer(&self) -> Vec<Vec<T>> {
-        let (channels, out_len) = self.get_max_output_size();
-        let mut wave_out = Vec::with_capacity(channels);
+        let frames = self.get_max_output_frames();
+        let channels = self.nbr_channels();
+        let mut buffer = Vec::with_capacity(channels);
         for _ in 0..channels {
-            wave_out.push(Vec::with_capacity(out_len));
+            buffer.push(Vec::with_capacity(frames));
         }
-        wave_out
+        buffer
     }
 
-    /// Get the max output size the resampler can give as (channels, frames).
+    /// Get the max number of output frames per channel.
     /// For the Resamplers with variable output sizes ([FftFixedIn] and [SincFixedIn]),
     /// the max output size is only guaranteed to be sufficient to prevent allocation
     /// within [Resampler::process_into_buffer] until the resampling ratio is changed.
-    fn get_max_output_size(&self) -> (usize, usize);
+    fn get_max_output_frames(&self) -> usize;
+
+    /// Get the maximum number of channels this Resampler can process
```

Comment:
I would change this to something like "Get the number of channels this resampler is configured for."